### DISCUSSION TITLE
Feature Request: A List of Missing Farming Fortune in API

### DISCUSSION THREAD

**Discussion by ptlthg** [HypixelDev/PublicAPI#618] [Ideas]
With the upcoming Garden v2 update and the lack of any Garden data being in the API yet, I thought it would be a good idea to throw together a list of what is currently missing that would be useful to expose in the API. My main motivation for this is to create free tools for players that use their farming fortune totals without needing manual input for [elitebot.dev](https://elitebot.dev), but of course it would be useful for everyone.

**EDIT:** Most of this was added on July 17th, 2024 in the new `/skyblock/garden` endpoint!

# Currently Missing
- [ ] Anita Talisman Buffed Jacob Contests
	- Using some deterministic method of selecting these would be great as long as that info is shared
- [ ] Temporary Fortune from Emptying Pest Bag
	- Would go in `temp_stat_buffs`, but not that necessary considering it's only 30 minutes
- [ ] Community Center Fortune Upgrade
	- A separate discussion I made has an idea on how to expose this without leaking monetization data #647

## Added (Thank you!)
- [x] Crop Milestone Progress
- [x] Crop Fortune Upgrades
- [x] Unlocked Plots
   - Just a count would be fine, knowing which plots are unlocked could be used for visualizations though
- [x] Garden XP
- [X] Exportable crop fortune from Carrolyn
	- Now visible via skyblock XP tasks. An actual property for these would be nice but it's fine now.
- [x] Unique Visitors
    - Currently can be inferred from the lore of Lotus equipment with Green Thumb, but this approach is flawed and annoying to use
- [x] Total Visitors Served
    - Can also be read from Lotus equipment, but caps at 5k until it turns into a `kills` field
- [x] Specific Visitor Stats
	- Times accepted/denied for each
	- This isn't a critical stat, but it would be nice to see
- [x] Composter Upgrades
	- Total compost generated would be a cool stat if it exists as well
- [x] Refined Dark Cocoa Truffles Consumed 
- [x] Wriggling Larva Consumed
	- Not sure where it should go, but this count should be somewhere
- [X] Pests Killed
	- Should be in bestiary, not worried about this being left out
- [x] Unique Medals Per Crop (diamond/platinum/gold/silver/bronze)
	- Should be able to be determined from contest history, but unique golds already exists and this would save on computation
- [x] Personal Best Jacob Scores
	- It's unclear how this will be determined (date cutoffs/etc), so having a field with the number will be a big help when calculating fortune
	
## Fortune Totals
I don't expect this to be added as I have no idea what the backend looks like for this system, but if the fortune totals as shown in alpha could be added as a number in the API, that would be amazing. It doesn't need to include tools/armor/equipment in the sum, as that depends on what the player has on them, but a sum of all the static numbers would be a nice addition. (Other stats being included would be nice as well)
	
 
## Calendar Endpoint
It's been requested a lot, but please consider adding a `/skyblock/calendar` endpoint. I had to construct my own system to get upcoming Jacob contests by relying on players who opted-in to sharing their calendar from SkyHanni. Currently, [my endpoint](https://api.elitebot.dev/contests/at/now) has served over 1 million requests to SkyHanni users alone in the span of a couple of months. Relying on crowdsourced data is inherently flawed and not something that should be encouraged, so please add an upcoming SkyBlock events endpoint.

Thank you admins for all the work you've done with the API lately!

**Comment by ptlthg** [comment]
It's been discovered that the Anita Talisman buffs different crops in each contest per player. This makes it kinda annoying to do anything with this information, and basically impossible to know if a player used it during a contest. If it could be made deterministic based off of the player UUID or synced between players, that would be a lot nicer for displaying and accounting for it.

**Comment by ptlthg** [comment]
Exportable carrots appear to be in the API under `[member].rift.village_plaza.cowboy.exported_carrots`, however this doesn't show up for people who didn't interact with this in game

I do have the +12 fortune but see this (since I haven't really played rift):
![image](https://github.com/user-attachments/assets/ef672b7a-422a-417a-a8b6-9c499a1de435)

And someone who has:
![image](https://github.com/user-attachments/assets/57e32243-d169-4530-a538-80db9f2cf442)

I'm not entirely sure if this is the correct data to be able to determine if someone has the fortune from Carrolyn, but it's something worth fixing anyway, seeing as the other exportable crops aren't available in any way.

**Comment by lunaynx** [comment]
Half-eaten Mushroom and Warty can also be added to the list of exportable crops.
